### PR TITLE
Webpack bundling changes

### PR DESCRIPTION
**Pull request by ivansg44**
`yarn build:web` takes a while, and rebuilding the entire application isn't necessary if users are only editing schemas (as bundling the schemas takes much less time than bundling everything else).

I introduce `yarn build:schemas`, which only updates the schema content and is pretty quick.

How does this work?:

* `menu`, `getSchema`, and `getExportFormats` are moved to `schemas.js`
* `index.js` imports these as `import ... from 'schemas'`
* We tell `webpack.config.js` that `schemas` is an external library
  * i.e., don't bundle `schemas.js`
* The new `webpack.schemas.js` file bundles `schemas.js` instead, and outputs a locally available library
* We import the path to this locally available `schemas` library inside `index.html`, which allows the original `webpack.config.js` build to reference this external `schemas` library without actually bundling it

Since the functions inside `schemas.js` are the only JavaScript that actually imports schema stuff (specifically `schema.json` and `export.js files`), any edits to schema stuff will not affect the `webpack.config.js` build. So if a user edits the schema content, they do not have to re-run the lengthy `webpack.config.js` build. Re-running `webpack.schemas.js` will suffice.

### CODE REVIEW

**Comment by ivansg44** [comment]
I need to continue testing and make sure I didn't break anything

**Comment by ivansg44** [comment]
TODO edit README?

File: web/webpack.config.js [hunk 1]
```diff
@@ -9,29 +9,17 @@ module.exports = (env, argv) => {
     output: {
       path: path.resolve(__dirname, 'dist'),
       filename: 'scripts/[name].js',
-      assetModuleFilename: 'assets/[hash][ext][query]',
+    },
+    externals: {
+      // Schemas lib external to this webpack build. Imported in html file.
```
Comment (ivansg44) [inline web/webpack.config.js:14]: More explicit: import has to be in JavaScript

Comment (ivansg44) [reply web/webpack.config.js:14]: Does it have to be in the header?

File: web/schemas.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import menu_ from './templates/menu.json';
+
+export const menu = menu_;
+export const getSchema = async (schema) => {
+  return (await import(`./templates/${schema}/schema.json`)).default;
+};
+export const getExportFormats = async (schema) => {
```
Comment (ivansg44) [inline web/schemas.js:7]: getLocale?

File: web/webpack.config.js [hunk 3]
```diff
@@ -61,8 +55,43 @@ module.exports = (env, argv) => {
...
+  // content was to reduce production build times when users are only editing
+  // schema files (and not the rest of the application), but the dev server
+  // already gets around this problem through hot loading.
   if (argv.mode === 'development') {
     config.devtool = 'eval-source-map';
+    // The schemas lib will come from a direct import of the javascript, instead
+    // of embedding an external lib into the html
+    config.resolve = {
+      alias: {
+        schemas: path.resolve(__dirname, 'schemas.js'),
```
Comment (ivansg44) [inline web/webpack.config.js:72]: Leave some comments saying this is what NML is going to have do for their thing

**Comment by ddooley** [comment]
Looks great, I'm merging into master!  And then Ken can merge into DH2-i18n-rc1 on his schedule.  The Wastewater update will also have to be merged into master and into DH2-i18n-rc1, but I can do this as soon as Charlie ok's it.

**Pull request merged**